### PR TITLE
[ticket/15942] Allow array in confirm_box title

### PR DESCRIPTION
**Pull request by mrgoldy**
PHPBB3-15942

Checklist:

- [x] Correct branch: master for new features; 3.2.x for fixes
- [x] Tests pass
- [x] Code follows coding guidelines: [master](https://area51.phpbb.com/docs/dev/master/development/coding_guidelines.html) and [3.2.x](https://area51.phpbb.com/docs/dev/3.2.x/development/coding_guidelines.html)
- [x] Commit follows commit message [format](https://area51.phpbb.com/docs/dev/3.2.x/development/git.html)

Tracker ticket (set the ticket ID to **your ticket ID**):

https://tracker.phpbb.com/browse/PHPBB3-15942

### CODE REVIEW

File: phpBB/includes/functions.php [hunk 1]
```diff
@@ -2130,7 +2130,7 @@ function check_form_key($form_name, $timespan = false)
 /**
 * Build Confirm box
 * @param boolean $check True for checking if confirmed (without any additional parameters) and false for displaying the confirm box
-* @param string $title Title/Message used for confirm box.
+* @param string|array $title Title/Message used for confirm box.
```
Comment (marc1706) [inline phpBB/includes/functions.php]: Please extend this to also state how the array version works, i..e how the array should be structured to work.

File: phpBB/includes/functions.php [hunk 2]
```diff
@@ -2182,13 +2182,27 @@ function confirm_box($check, $title = '', $hidden = '', $html_body = 'confirm_bo
 	// generate activation key
 	$confirm_key = gen_rand_string(10);
 
+	// generate language strings
+	if (is_array($title))
+	{
+		$key = array_shift($title);
+		$count = array_shift($title);
+		$confirm_title = isset($user->lang[$key]) ? $user->lang($key, $count, $title) : $user->lang('CONFIRM');
```
Comment (VSEphpbb) [inline phpBB/includes/functions.php]: Should probably switch from user->lang to language->lang while we're at it.

File: phpBB/includes/compatibility_globals.php [hunk 3]
```diff
@@ -48,6 +48,9 @@ function register_compatibility_globals()
 	/* @var $user \phpbb\user */
 	$user = $phpbb_container->get('user');
 
+	/* @var \phpbb\language\language $language */
+	$language = $phpbb_container->get('language');
```
Comment (3D-I) [inline phpBB/includes/compatibility_globals.php]: Nothing relevant, though I'd put this one before `$phpbb_log = $phpbb_container->get('log');`

Comment (marc1706) [reply phpBB/includes/compatibility_globals.php]: I put it next to `$user` for historic reasons.

File: phpBB/includes/compatibility_globals.php [hunk 4]
```diff
@@ -29,7 +29,7 @@ function register_compatibility_globals()
 {
 	global $phpbb_container;
 
-	global $cache, $phpbb_dispatcher, $request, $user, $auth, $db, $config, $phpbb_log;
+	global $cache, $phpbb_dispatcher, $request, $user, $auth, $db, $config, $language, $phpbb_log;
```
Comment (3D-I) [inline phpBB/includes/compatibility_globals.php]: Yeah nothing relevant as I said, then for the same reason `$language` should go after `$user` in this line. ;)
Again, it's not important nor relevant. Thanks.

File: phpBB/includes/compatibility_globals.php (hunk 3 above)
Comment (3D-I) [reply phpBB/includes/compatibility_globals.php]: https://github.com/phpbb/phpbb/pull/5513/files/0e38bebaa56dcf8921df99e8eb85e3046721dd13#r276113152

**Pull request merged**